qpicard/src/org/qcmg/picard/util/BAMFileUtils.java

```diff
@@ -70,16 +73,20 @@ public static void renameBamIndex(File output) throws IOException{
 	    final Path indexqcmg =  IOUtil.addExtension(Paths.get(path), FileExtensions.BAI_INDEX);
 		 			
 		//rename files
-		Files.move(indexpicard, indexqcmg, StandardCopyOption.REPLACE_EXISTING);				
+		Files.move(indexpicard, indexqcmg, StandardCopyOption.REPLACE_EXISTING);
+		
 	}
 	/**
 	 * picard index name created by adding .bai to .cram. But we want to have index name end with .cram.crai
 	 * @param cramFile
 	 * @throws Exception 
 	 */	
 	
-	public static void renameCramIndex(File output) throws IOException{
-		String path = output.getPath();
+	public static void renameCramIndex(File input) throws IOException{
+		
+		if( input == null ) throw new IOException("we can't rename a NULL file!");
```

Comment:
IllegalArgumentException rather that IOException

Reply:
yes, I agree It is an Argument Exception but here this argument is also IO related.  Actually, it is a NullPointerException but we just want to handle it and throw more related error message.

Reply:
Check out the javadocs: https://docs.oracle.com/javase/8/docs/api/java/lang/IllegalArgumentException.html